packages/features/auth/lib/onboardingUtils.ts

```diff
@@ -0,0 +1,78 @@
+import dayjs from "@calcom/dayjs";
+import { FeaturesRepository } from "@calcom/features/flags/features.repository";
+import prisma from "@calcom/prisma";
+
+const ONBOARDING_INTRODUCED_AT = dayjs("September 1 2021").toISOString();
+
+/**
+ * Checks if a user needs onboarding on the server side.
+ * Returns the onboarding path if redirect is needed, null otherwise.
+ *
+ * @param userId - The user ID to check
+ * @param options - Optional configuration
+ * @param options.checkEmailVerification - Whether to check if email verification is required
+ * @param options.organizationId - Optional organizationId from session (to avoid extra query)
+ */
+export async function checkOnboardingRedirect(
+  userId: number,
+  options?: {
+    checkEmailVerification?: boolean;
+    organizationId?: number | null;
+  }
+): Promise<string | null> {
+  // Query minimal user data needed for onboarding check
+  const user = await prisma.user.findUnique({
+    where: { id: userId },
+    select: {
+      completedOnboarding: true,
+      createdDate: true,
+      emailVerified: true,
+      identityProvider: true,
+    },
+  });
+
+  if (!user) {
+    return null;
+  }
```

Comment:
Not focusing on abstracting this out of a repo just yet - its purly a copy of an existing function that we can reuse